public/src/pages/views/MemberView.tsx

```diff
@@ -123,44 +123,48 @@ export class MemberView extends React.Component<IUserViewerProps, IUserViewerSta
         </div>;
     }
 
-    public handleAction(userRel: IUserRelation, link: ILink) {
+    public async handleAction(userRel: IUserRelation, link: ILink) {
```

Comment:
This code was a bit long. Could these be split into separate methods for the different cases?

Reply:
Done.